IdentityCore/src/cache/key/MSIDDefaultCredentialCacheKey.m

```diff
@@ -99,6 +106,14 @@ - (instancetype)initWithHomeAccountId:(NSString *)homeAccountId
         _environment = environment;
         _clientId = clientId;
         _credentialType = type;
+        
+        if (_credentialType == MSIDAccessTokenType)
+        {
+            _enrollmentId = [[MSIDIntuneEnrollmentIdsCache sharedCache] enrollmentIdForHomeAccountId:homeAccountId
+                                                                                        legacyUserId:nil
```

Comment:
should we also lookup by legacy user ID as we did for requests? Otherwise we might not find correct enrollment ID.

Reply:
Investigating if this is doable - need to make sure this is covered for all accessToken operations (save, get, remove) across entire codebase, for both CredentialCacheKeys and CredentialCacheQueries.

Reply:
Discussed offline, adding enrollmentId field to credentialCacheItem(non-nil for access tokens only) and MSIDAccessToken.